demos/gallery/samples/common/CanvasWidget.js

```diff
@@ -0,0 +1,32 @@
+import { CanvasWidget } from "@hpcc-js/common";
+
+const data = [
+    ["A", 100],
+    ["B", 162],
+    ["C", 262],
+];
+var w = new CanvasWidget().target("target").data(data);
+const origEnter = w.enter;
+w.enter = function (domNode, element) {
+    origEnter.apply(this, arguments);
+    this.resize();
+};
+const origUpdate = w.update;
+w.update = function (domNode, element) {
+    origUpdate.apply(this, arguments);
+    const gutter = 4;
+    const rowHeight = 30;
+    const labelWidth = 10;
+    this._ctx = domNode.getContext("2d");
+    this.data().forEach((row,i)=>{
+        this._ctx.fillStyle = "red";
+        this._ctx.fillText(row[0], gutter, (rowHeight * i) + (gutter * (i+1)) + (rowHeight/2));
+        this._ctx.fillRect(
+            gutter + labelWidth + gutter, 
+            (rowHeight * i) + (gutter * (i+1)),
+            row[1], 
+            rowHeight - gutter
+        );
+    })
+};
+w.render();
```

Comment:
Change this to be a new `class` which derived from CanvasWidget with that update code:
```
class MyCanvasWidget extends CanvasWidget {
...
}
const w = new MyCanvasEidget().target(...)
```
Type of thing